__tests__/app.test.js

```diff
@@ -24,141 +24,192 @@ describe("/api", () => {
+        expect(response.body.message).toBe("endpoint not found!");
+      });
+  });
+});
+describe("get /api/articles/:article_id", () => {
+  test(" GET /api/articles/1 returns a status 200 and a single corresponding article", () => {
+    return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then(({ body }) => {
+        const article = body.article[0];
+        expect(article.article_id).toBe(1);
+        expect(article.title).toBe("Living in the shadow of a great man");
+        expect(article.topic).toBe("mitch");
+        expect(article.author).toBe("butter_bridge");
+        expect(article.body).toBe("I find this existence challenging");
+        expect(article.created_at).toBe("2020-07-09T20:11:00.000Z");
+        expect(article.votes).toBe(100);
+        expect(article.article_img_url).toBe(
+          "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700"
+        );
+      });
+  });
+  test("GET /api/articles/banana returns a 404 and 'not found' as parameter is invalid", () => {
+    return request(app)
+      .get("/api/articles/hello")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.message).toBe("bad request");
+      });
+  });
+  test("GET /api/articles/100 returns a 404 and error message as parameter is valid but doesn't exist in database", () => {
+    return request(app)
+      .get("/api/articles/100")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.message).toBe("not found");
+      });
+  });
+});
+describe("GET /api/articles/:article_id/comments", () => {
+  test("returns an array", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then(({ body }) => {
+        expect(Array.isArray(body)).toBe(true);
+      });
+  });
+  test("returned array has desired properties", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then(({ body }) => {
+        body.forEach((comment) => {
+          expect(comment.hasOwnProperty("comment_id")).toBe(true);
+          expect(comment.hasOwnProperty("votes")).toBe(true);
+          expect(comment.hasOwnProperty("created_at")).toBe(true);
+          expect(comment.hasOwnProperty("author")).toBe(true);
+          expect(comment.hasOwnProperty("body")).toBe(true);
+          expect(comment.hasOwnProperty("article_id")).toBe(true);
         });
-    });
+      });
+  });
+  test("returned array is ordered by most recent comments first", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then(({ body }) => {
+        expect([body]).toBeSortedBy("created_at");
+      });
+  });
+  test("invalid article id sends 400 error and message", () => {
+    return request(app)
+      .get("/api/articles/banana/comments")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.message).toBe("bad request");
```

Comment:
what if its a 200 but theres no comments yet?

Reply:
Hi Pippa! I've sorted my code taking into account your other comments but I'm struggling with this one...will there ever be a 200 and no comments? the endpoint is to find the comments for that article, so my thinking is that there wouldn't be a 200 and no comments as the endpoint is meant to take the user to the comments, so wouldn't it just be a 404...?

Reply:
if the article exists but doesnt have any comments itll be 200 but if like below the article doesnt exist thats a 404